MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatSetup.cs

```diff
@@ -11,12 +11,20 @@
 using MvvmCross.Droid.Support.V4;
 using MvvmCross.Platforms.Android.Core;
 using MvvmCross.Platforms.Android.Presenters;
+using MvvmCross.Presenters;
 using MvvmCross.ViewModels;
 
 namespace MvvmCross.Droid.Support.V7.AppCompat
 {
     public abstract class MvxAppCompatSetup : MvxAndroidSetup
     {
+        protected override void InitializeIoC()
+        {
+            base.InitializeIoC();
+
+            Mvx.LazyConstructAndRegisterSingleton<IMvxViewPresenter, MvxAppCompatViewPresenter>();
```

Comment:
It looks a bit weird to register all ViewPresenters from `InitializeIoC`... Although I suppose this is because we need it registered as early as possible (right?). 

Maybe we can at least create a separate method for that? That would make overriding the behavior cleaner

Reply:
I get you're point - the InitializeIoC doesn't indicate that it's going to register the view presenter. However, if we create methods for anything we register, we'll end up with a ton of overridable methods. imho it seems logical that all registration is done as part of initializing the IoC container.
I guess we could have a single overrideable method which is RegisterLazySingletons (or something like that) - it needs to indicate lazy because there's no guarrantee at the point this is done that all implementations will be available.